packages/react-select/src/animated/transitions.tsx

```diff
@@ -72,93 +66,62 @@ interface CollapseProps {
 
     /*
-      A check on existence of ref should not be necessary at this point,
-      but TypeScript demands it.
+      Here we're invoking requestAnimationFrame with a callback invoking our
+      call to getBoundingClientRect and setState in order to resolve an edge case
+      around portalling. Certain portalling solutions briefly remove children from the DOM
+      before appending them to the target node. This is to avoid us trying to call getBoundingClientrect
+      while the Select component is in this state.
     */
-    if (ref) {
-      /*
-        Here we're invoking requestAnimationFrame with a callback invoking our
-        call to getBoundingClientRect and setState in order to resolve an edge case
-        around portalling. Certain portalling solutions briefly remove children from the DOM
-        before appending them to the target node. This is to avoid us trying to call getBoundingClientrect
-        while the Select component is in this state.
-      */
-      // cannot use `offsetWidth` because it is rounded
-      this.rafID = window.requestAnimationFrame(() => {
-        const { width } = ref.getBoundingClientRect();
-        this.setState({ width });
-      });
-    }
-  }
-
-  componentWillUnmount() {
-    if (this.rafID) {
-      window.cancelAnimationFrame(this.rafID);
-    }
-  }
-
-  // get base styles
-  getStyle = (width: Width): CSSProperties => ({
-    overflow: 'hidden',
-    whiteSpace: 'nowrap',
-    width,
-  });
-
-  // get transition styles
-  getTransition = (state: TransitionStatus) => this.transition[state];
-
-  render() {
-    const { children, in: inProp, onExited } = this.props;
-    const exitedProp = () => {
-      if (this.nodeRef.current && onExited) {
-        onExited(this.nodeRef.current);
-      }
-    };
+    // cannot use `offsetWidth` because it is rounded
+    const rafId = window.requestAnimationFrame(() =>
+      setWidth(current.getBoundingClientRect().width)
+    );
 
-    const { width } = this.state;
+    return () => window.cancelAnimationFrame(rafId);
+  }, []);
 
-    return (
-      <Transition
-        enter={false}
-        mountOnEnter
-        unmountOnExit
-        in={inProp}
-        onExited={exitedProp}
-        timeout={this.duration}
-        nodeRef={this.nodeRef}
-      >
-        {(state) => {
-          const style = {
-            ...this.getStyle(width),
-            ...this.getTransition(state),
-          };
-          return (
-            <div ref={this.nodeRef} style={style}>
-              {children}
-            </div>
-          );
-        }}
-      </Transition>
-    );
-  }
-}
+  return (
+    <Transition
+      enter={false}
+      mountOnEnter
+      unmountOnExit
+      in={_in}
+      onExited={() => {
+        const { current } = ref;
+        if (!current) return;
+        onExited?.(current);
+      }}
+      timeout={collapseDuration}
+      nodeRef={ref}
+    >
+      {(state) => (
+        <div
+          ref={ref}
+          style={{
```

Comment:
Nit: Could the nested ternaries be against an object with state keys, or some `getStyleFromState` fn with a switch?